sdks/java/io/cassandra/src/test/java/org/apache/beam/sdk/io/cassandra/CassandraIOTest.java

```diff
@@ -126,9 +126,11 @@ public static void startCassandra() throws Exception {
 
   @AfterClass
   public static void stopCassandra() {
-    EmbeddedCassandraServerHelper.cleanEmbeddedCassandra();
-    session.close();
-    cluster.close();
+    if (cluster != null && session != null) {
+      EmbeddedCassandraServerHelper.cleanEmbeddedCassandra();
+      session.close();
+      cluster.close();
```

Comment:
Would break this into two if blocks, depending on where we fail, one of them might be non null.

Reply:
well, `session = cluster.connect().` So if cluster is null (because there was an exception in startup), then session is null.  And if not, cluster.connect() is a blocking call to connectAsync future.  That is why I wrote that way. Do you want me to change it ?

Reply:
I was worried that if there is an exception creating the session, then the cluster won't be shutdown.

Reply:
Ok for me to keep it that way.

Reply:
Good point, I guess it can happen that there is an exception in session creation but have the cluster well created, I'll add the code.